python/tests/job_runner/test_job_dispatch.py

```diff
@@ -124,3 +124,166 @@ def test_job_dispatch_kills_itself_after_unsuccessful_job(self):
+                job_dispatch_script,
+                current_dir,
+            ]
+        )
+
+        job_dispatch_process.wait()
+
+        assert not os.path.isfile("{}/event_log".format(current_dir))
+
+    @tmpdir()
+    def test_job_runner_cli_event_reporting_enabled(self):
+
+        with open("dummy_executable", "w") as f:
+            f.write(
+                "#!/usr/bin/env python\n"
+                "import sys, os\n"
+                'filename = "job_{}.out".format(sys.argv[1])\n'
+                'f = open(filename, "w")\n'
+                "f.close()\n"
+            )
+
+        executable = os.path.realpath("dummy_executable")
+        os.chmod("dummy_executable", stat.S_IRWXU | stat.S_IRWXO | stat.S_IRWXG)
+
+        self.job_list = {
+            "umask": "0002",
+            "DATA_ROOT": "",
+            "global_environment": {},
+            "global_update_path": {},
+            "ee_id": "_ID_",
+            "jobList": [
+                {
+                    "name": "job_A",
+                    "executable": executable,
+                    "target_file": None,
+                    "error_file": None,
+                    "start_file": None,
+                    "stdout": "dummy.stdout",
+                    "stderr": "dummy.stderr",
+                    "stdin": None,
+                    "argList": ["A"],
+                    "environment": None,
+                    "exec_env": None,
+                    "license_path": None,
+                    "max_running_minutes": None,
+                    "max_running": None,
+                    "min_arg": 1,
+                    "arg_types": [],
+                    "max_arg": None,
+                },
+            ],
+            "run_id": "",
+            "ert_pid": "",
+        }
+
+        with open("jobs.json", "w") as f:
+            f.write(json.dumps(self.job_list))
+
+        current_dir = os.path.realpath(os.curdir)
+
+        # Required to execute job_dispatch in separate process group by
+        # os.setsid moves the current process out of the current group
+        with open("job_dispatch_executer", "w") as f:
+            f.write(
+                "#!/usr/bin/env python\n"
+                "import os, sys\n"
+                "os.setsid()\n"
+                "os.execv(sys.argv[1], sys.argv[1:])\n"
+                "\n"
+            )
+        os.chmod("job_dispatch_executer", stat.S_IRWXU | stat.S_IRWXO | stat.S_IRWXG)
+
+        job_dispatch_script = os.path.realpath(
+            os.path.join(
+                os.path.dirname(os.path.abspath(__file__)),
+                "../../../bin/job_dispatch.py",
+            )
+        )
+
+        job_dispatch_process = Popen(
+            [
+                os.path.realpath("job_dispatch_executer"),
+                job_dispatch_script,
+                current_dir,
+            ]
+        )
+
+        job_dispatch_process.wait()
+
+        assert os.path.isfile("{}/event_log".format(current_dir))
```

Comment:
These tests seems to be near copies of `test_terminate_jobs`? Maybe this duplication could be reduced. Or maybe the assert that the event reporter is absent is moved to `test_terminate_jobs`.

Or, the creation of reporters is moved out of `main` into a function, which could then be unit tested.

Reply:
There are several options here. Code duplication could be reduced for sure. There is a lot of common code with the tests in master for the argparser. I will look in to this